### ISSUE TITLE
Increase Stamina/HP/Range per point

### ISSUE THREAD

**Issue by SoulMuncher** [jdoleary/Spellmasons#86]
Putting stat points into mana is _infinitely_ more powerful than putting stat points elsewhere.

Mana is better at giving you endurance than health is.
20 mana or 32 health (less health than most enemies do damage per-attack)
- This is an extra blood curse, purify, heal, shield, and similar spell every turn.
- This is the difference between being able to resurrect the 1000 hp miniboss or tanking the entire horde of enemies yourself
- You can turn it into offense, to kill the enemy that would've otherwise hit you
- Decoy.

Mana is better at giving you mobility than stamina is. 
20 mana or 40 stamina (19% walking distance):
- This is an extra push, pull, repel, dash, swap to move somewhere
- This is an extra targeting spell to reach something
- This is an extra slash to slow, freeze, or kill the enemy chasing you

Mana is better at giving you range than cast range is:
20 mana or 12% cast range
- This is an extra cone, column, or even plus radius
- This is the push before you connect
- This is what lets you add target similar to your mid game spell to wipe out all the summoners

It is individually better at doing everything the other stats can do, while also having the versatility to switch between uses as the situation changes, AND it gives you more damage. It lets you kill Deathmason turn 1 and then target similar + debilitate + rend when he splits into 3.

On top of that, Mana is the only stat that doesn't have diminishing returns, and in fact has increasingly exponential returns. Every point of health, stamina, or range increases that stat by a smaller amount, and affects the game less than the last point did, while every 20 mana is another targeting spell, which often doubles your potential for damage, health, range, number of summons, or turn efficiency (letting you do that big combo all at once instead of over two turns)

-

While a huge part of this imbalance is due to the multiplicative nature of targeting spells, I do also think that the raw stat gains per point need significant adjustment.

In the long run, and alongside a targeting spell nerf, it might also be a good idea to increase the player's starting mana, so they aren't immediately forced to take it.

**Comment by jdoleary** [comment]
Haha totally agree.  Let's definitely buff the stam/hp/range per point. 

As for the targeting nerf, I want to see how the changes I'm making to Target Similar and Target Kind affect that.  It'll be interesting to see if that makes "map-wipe" spells less common.  We can adjust gradually.

**Comment by SoulMuncher** [comment]
@jdoleary 
I've been doing some testing, here are my initial thoughts:

HP: +20 hp per point feels about right. One full level lets me tank a couple extra hits, which can definitely come in handy. I don't foresee myself going over 100 hp in a serious run, but it's still more interesting than taking 1 point for one-shot-protection and calling it a day. Especially when looking at spells that cost health.

Stamina: I tried a few different values here. Less than +50 stamina per point feels like I should be taking mana instead. More than +50 stamina per stat point feels... A little absurd. Not powerful, just silly. I'm going to go ahead and put it at 25 (maybe 20, just to be consistent with other stats), and wait for broader mana rebalancing to adjust further. I don't feel like this is ever worth taking over mana. I'm not sure what to do with this stat.

Cast Range: I lowered the starting value from 240 to 200, and set it to +20 per point. I really like this change, because it always feels like a viable option without being too strong.

Note: Consider how this makes some classes stronger or weaker.
? -> Buffs Spellmason and Necro | Nerfs Cleric

**Comment by SoulMuncher** [comment]
Thought I'd quickly share an idea I had, free to close if you disagree

We reverted the cast range change to prevent making the early game too difficult for new players. What if we increased the starting hp to 60 instead, and kept 200 base attack range?

Reason/Effect:
- 20 more health is difference between dying in one turn on level 1 or living to cast another slash. It will also allow you to survive a hit from mid game blood golems and archers. This is much more likely to help a new player through the early stages of the game, and much more forgiving overall.
- A new player is likely to die due to the mismanagement of mana or stamina, or a misunderstanding of game mechanics. Cast range usually won't help with this, whereas an extra turn will.
- 20 more health makes early game significantly easier, while not being as impactful in endgame. I feel that Cast Range is opposite in that it's value shines most during late game.
- Would cause visual cast range to go up by 10 instead of 8,8,9
- Makes it harder for Farseer to outrange everything in the game
- Cast Range feels a little more rewarding to level up

**Comment by jdoleary** [comment]
Yeah I like this reasoning, it would also make it so you kinda have to upgrade cast range at some point or else you have such little range.  It makes you more likely to consider non mana upgrades.

Note: Stat upgrades don't start appearing until level 2 I think (or is that just for the tutorial run, I don't remember).  If we do this I think it should show after every level
